### PR TITLE
DROOLS-1733 Enhance test coverage of FEEL built-in functions

### PR DESCRIPTION
**Pull request by baldimir**
Master PR here: https://github.com/kiegroup/drools/pull/1474

@etirelli @tarilabs could you please review?

### CODE REVIEW

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/AnyFunction.java [hunk 1]
```diff
@@ -33,15 +33,15 @@ public AnyFunction() {
...
-            return FEELFnResult.ofError(new InvalidParametersEvent(Severity.ERROR, "an element in the list is not a Boolean"));
-        }
+        boolean result = false;
         for ( final Object element : list ) {
-            if (element != null && element == Boolean.TRUE) {
-                return FEELFnResult.ofResult( true );
+            if (element != null && !(element instanceof Boolean)) {
+                return FEELFnResult.ofError(new InvalidParametersEvent(Severity.ERROR, "an element in the list is not a Boolean"));
+            } else {
+                result |= element != null && element == Boolean.TRUE;
```
Comment (etirelli) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/AnyFunction.java]: @baldimir Need a small correction here. The spec says: 
> return true if any item is true, else false if all items are false, else null

The previous implementation was wrong, but good opportunity to fix.

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/AllFunction.java [hunk 2]
```diff
@@ -33,15 +32,15 @@ public AllFunction() {
...
-            return FEELFnResult.ofError(new InvalidParametersEvent(Severity.ERROR, "an element in the list is not a Boolean"));
-        }
+        boolean result = true;
         for ( final Object element : list ) {
-            if (element == null || element == Boolean.FALSE) {
-                return FEELFnResult.ofResult( false );
+            if (element != null && !(element instanceof Boolean)) {
+                return FEELFnResult.ofError(new InvalidParametersEvent(Severity.ERROR, "an element in the list is not a Boolean"));
+            } else {
+                result &= element != null && element == Boolean.TRUE;
```
Comment (etirelli) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/AllFunction.java]: We need a small correction here. Spec says:
> return false if any item is false, else true if all items are true, else null

The previous implementation was wrong, but good opportunity to fix.

**Comment by baldimir** [comment]
@etirelli AnyFunction and AllFunctions should be aligned now with the spec. Please review.

**Review by etirelli — APPROVED** [review]
Good to merge! Thank you @baldimir !

**Pull request merged**